src/test/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProviderIT.java

```diff
@@ -18,168 +18,125 @@
-import org.junit.runners.Parameterized;
-import org.mockito.Mockito;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 
-@RunWith(Parameterized.class)
 public class StreamingClientProviderIT {
-  private final boolean enableClientOptimization;
-  private final Map<String, String> clientConfig = TestUtils.getConfForStreaming();
 
-  @Parameterized.Parameters(name = "enableClientOptimization: {0}")
-  public static Collection<Object[]> input() {
-    return Arrays.asList(new Object[][] {{true}, {false}});
+  @Test
+  public void getClient_forOptimizationEnabled_returnSameClient() {
+    // given
+    Map<String, String> clientConfig = getClientConfig(true);
+    StreamingClientProvider streamingClientProvider =
+        StreamingClientProvider.getStreamingClientProviderInstance();
+
+    // when
+    SnowflakeStreamingIngestClient client = streamingClientProvider.getClient(clientConfig);
+    SnowflakeStreamingIngestClient client2 = streamingClientProvider.getClient(clientConfig);
+
+    // then
+    Assertions.assertNotNull(client);
+    Assertions.assertNotNull(client2);
+    Assertions.assertEquals(client, client2);
+
+    int clientsRegistered = streamingClientProvider.getRegisteredClients().size();
+    Assertions.assertEquals(1, clientsRegistered);
   }
 
-  public StreamingClientProviderIT(boolean enableClientOptimization) {
-    this.enableClientOptimization = enableClientOptimization;
-    this.clientConfig.put(
-        SnowflakeSinkConnectorConfig.ENABLE_STREAMING_CLIENT_OPTIMIZATION_CONFIG,
-        String.valueOf(this.enableClientOptimization));
+  @Test
+  public void getClient_forOptimizationDisabled_returnDifferentClients() {
+    // given
+    Map<String, String> clientConfig = getClientConfig(false);
+    StreamingClientProvider streamingClientProvider =
+        StreamingClientProvider.getStreamingClientProviderInstance();
+
+    // when
+    SnowflakeStreamingIngestClient client = streamingClientProvider.getClient(clientConfig);
+    SnowflakeStreamingIngestClient client2 = streamingClientProvider.getClient(clientConfig);
+
+    // then
+    Assertions.assertNotNull(client);
+    Assertions.assertNotNull(client2);
+    Assertions.assertNotEquals(client, client2);
+  }
+
+  @Test
+  public void getClient_forInvalidClient_returnValidClient() throws Exception {
+    // given
+    Map<String, String> clientConfig = getClientConfig(true);
+    StreamingClientProvider streamingClientProvider =
+        StreamingClientProvider.getStreamingClientProviderInstance();
+
+    // when
+    SnowflakeStreamingIngestClient client = streamingClientProvider.getClient(clientConfig);
+    client.close();
+    SnowflakeStreamingIngestClient client2 = streamingClientProvider.getClient(clientConfig);
+
+    // then
+    Assertions.assertTrue(client.isClosed());
+    Assertions.assertFalse(client2.isClosed());
```

Comment:
Will it make sense to verify that `client` and `client2` is the same client? It can be done by using `assertSame`.

Reply:
Actually in this case clients are **not** the same. But I switched to `assertSame` in the first test case.